src/NBitcoin/UInt256.cs

```diff
@@ -275,8 +289,11 @@ public override bool Equals(object obj)
             var item = obj as uint256;
             if (item == null)
                 return false;
+            
+            if (this.pn0 != item.pn0)
```

Comment:
I decided to put this in as its 5% performance improvement in release mode and 30% in debug

Reply:
how does that translate to actual difference it time?

Reply:
Negligible in real world 5ms on 1m comparisons, but debug it was a big difference ~40ms so useful while developing I guess